### ISSUE TITLE
Error when using Group EH "UnitKilled"

### ISSUE THREAD

**Issue by PiG13BR** [acemod/ACE3#10433]
**Mods (complete and add to the following information):**
- **Arma 3:** `2.18 stable`
- **CBA:** `3.18.1 stable`
- **ACE3:** `3.18.1 stable`

**Description:**
SP/editor - clean mission: Error in `.../addons/medical_feedback/XEH_postInit.sqf` when using group EH "UnitKilled". 
In dedi server .rpt: I've also got an error in the `fnc_setDead.sqf`. In this case was not a clean mission.
More info in the logs below

**Steps to reproduce:**
1. Open the editor in VR, place the player unit and some other unit. 
2. In the group init field of that unit, add the ["UnitKilled" EH](https://community.bistudio.com/wiki/Arma_3:_Event_Handlers#UnitKilled).
3. Preview the scenario and kill the unit.

**Expected behavior:**
No error showing up.

**Where did the issue occur?**
- Dedicated / Singleplayer / Editor (Singleplayer)

**Log Files:**
- [Client rpt file](https://pastebin.com/Vta2jEP7)
- [Server rpt file](https://pastebin.com/Lm4EtgV5)

**Additional context:**
- I've tested it out with "killed" EH and it works normally. Seems like the ace functions is getting the `_group parameter` from the "UnitKilled" EH (?), and this generates the error.
- Also adding the EH in the init field of that unit (e.g. (group this) addEventHandler...) generates the same issue.
- Same issue if placing more than one unit in that group.

**Screenshots:**
[Screenshots of the error after killing the unit with the group EH](https://imgur.com/a/AxfQVrz).
*Thanks to artemoz*

**Comment by johnb432** [comment]
Given that CBA is also throwing an error, I doubt this is ACE's doing - I doubt it's even CBA's fault. I'll test it myself, but I have a feeling that it's something in 2.18 that's been changed.

**Comment by PabstMirror** [comment]
ace_medical_death event is running for the object and then the group

**Comment by PiG13BR** [comment]
I think it's the EH fault. It's returning wrong parameters, at least, wrong order of them.
![image](https://github.com/user-attachments/assets/0118f04f-6c91-4236-a2d5-b12be601cc93)
The _unit parameter should return the killed unit but it's actually returning the _killer.

**Comment by johnb432** [comment]
ACE is not the culprit, as it still happens with CBA only:

```
20:55:16 Error in expression <ariable ["cba_xeh_killedBody", objNull] != _unit) then {                    _uni>
20:55:16   Error position: <!= _unit) then {                    _uni>
20:55:16   Error Generic error in expression
20:55:16  ➥ Context: 	[] L1 ()
	[] L1 ()
```

**Comment by PabstMirror** [comment]
It's really just BIS's fault 

when adding a "UnitKilled" EH to a group:
- Unit Killed XEH runs first and is passed 5 args
`[B Alpha 1-2,X2,X1,X1,true]` 1st is the group (these are the args that unitKilled group eh should be getting)

- UnitKilled runs 2nd and is passed
`[X2,X1,X1,true]` 1st arg is the unit (these are the args that the killed unit eh should be getting)

so for some reason adding the group event switches the args around 🤡

**Comment by johnb432** [comment]
Yea, confirmed that it happens in Vanilla only too.

**Comment by johnb432** [comment]
https://feedback.bistudio.com/T185567

**Comment by PiG13BR** [comment]
Seems like this issue was fixed in 2.18 hotfix.
https://dev.arma3.com/post/spotrep-00116

Tested myself with ace + CBA. No errors. Also checked .rpt file.
